export-w3c-test-changes creates duplicate branches instead of updating pre-existing export branch

briannafan opened the pull request:
#### 676ffb9985a4ac19525b059ba87ece65c6c41dad
<pre>
export-w3c-test-changes creates duplicate branches instead of updating pre-existing export branch
<a href="https://bugs.webkit.org/show_bug.cgi?id=296656">https://bugs.webkit.org/show_bug.cgi?id=296656</a>
<a href="https://rdar.apple.com/156960581">rdar://156960581</a>

Reviewed by Tim Nguyen.

Before 296936@main, the branch name got a suffix if `git show-ref --verify --quiet {branch_name}` == 0.
However, the branch name would never match with the ref since it didn&apos;t have the &quot;refs/heads&quot; prefix.
296936@main fixed this by using `branches_for`, but the duplicate branching is actually not what we want.

Having an existing branch get updated with new changes is the desired behaviour, so this PR removes the
logic in `_ensure_new_branch_name` and simply creates a branch name with the `wpt-export-for-webkit-` prefix.

* Tools/Scripts/webkitpy/w3c/test_exporter.py:
(WebPlatformTestExporter._branch_name):
(WebPlatformTestExporter._ensure_wpt_repository):
(WebPlatformTestExporter.create_wpt_pull_request):
(WebPlatformTestExporter._ensure_new_branch_name): Deleted.

Canonical link: <a href="https://commits.webkit.org/298053@main">https://commits.webkit.org/298053@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2aaca5e943c7e50580e09346ac5300291e329852

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114090 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33842 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24300 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120258 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/64854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b764855b-bd9e-48bc-b8c0-6ac3b0d04e36) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115979 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34470 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42402 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/86709 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/64854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6f5fff8b-cad3-4b8a-80d1-a2887e543d17) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117038 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/27443 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/102471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/67093 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/113464 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/26626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/20599 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/63957 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/96810 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/20716 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/123483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/41118 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/30639 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/123483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/41497 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/98680 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/123483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/18275 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/37264 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18287 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/40996 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/40616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/43916 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/42369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Tools/Scripts/webkitpy/w3c/test_exporter.py:
```diff
@@ -125,7 +125,7 @@ def _wpt_fork_push_url(self):
     @memoized
     def _branch_name(self):
-        return self._ensure_new_branch_name()
+        return f"wpt-export-for-webkit-{(str(self._bug_id) if self._bug_id else '0')}"
```
Maybe the git commit revision makes more sense as fallback than "0", or throwing an error if there's no bug_id

nt1m reviewed (approved):
Could we have an unit test for it?

briannafan committed:
export-w3c-test-changes creates duplicate branches instead of updating pre-existing export branch
https://bugs.webkit.org/show_bug.cgi?id=296656
rdar://156960581

Reviewed by Tim Nguyen.

Before 296936@main, the branch name got a suffix if `git show-ref --verify --quiet {branch_name}` == 0.
However, the branch name would never match with the ref since it didn't have the "refs/heads" prefix.
296936@main fixed this by using `branches_for`, but the duplicate branching is actually not what we want.

Having an existing branch get updated with new changes is the desired behaviour, so this PR removes the
logic in `_ensure_new_branch_name` and simply creates a branch name with the `wpt-export-for-webkit-` prefix.

* Tools/Scripts/webkitpy/w3c/test_exporter.py:
(WebPlatformTestExporter._branch_name):
(WebPlatformTestExporter._ensure_wpt_repository):
(WebPlatformTestExporter.create_wpt_pull_request):
(WebPlatformTestExporter._ensure_new_branch_name): Deleted.

Canonical link: https://commits.webkit.org/298053@main
